### ISSUE TITLE
Encoder for bytes field gets incorrect argument value

### ISSUE THREAD

**Issue by tstehr** [lidatong/dataclasses-json#144]
I'm trying to build a dataclass with a field of type `bytes` and have it encoded as base64 in JSON. To archive that, I have defined a custom encoder:

```python
@dataclass_json
@dataclass(frozen=True)
class TestCase:
    byte_field: bytes = field(metadata=config(
        encoder=lambda val: base64.b64encode(val).decode(),
        mm_field=fields.String()
    ))

t = TestCase(byte_field=b'abc')
print(t.to_json())
```

However when executing this the encoder is not called with `b'abc'` as expected, but with `[97, 98, 99]` (the int representation of the bytes) instead.

I believe this happens because `dataclasses_json.core._asdict` treats the byte string as a collection and converts it to a list of int.

**Comment by tstehr** [comment]
I have submitted #145 as a possible fix

**Comment by lidatong** [comment]
Fixed by #145
